File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        # Returns the product of the digits of a number.
        def prod(x):
            # Single-digit numbers are their own digit product.
            if x <= 9:
                return x

            product = 1
            while x != 0:
                product *= (x % 10)
                x //= 10

            return product

        # If t is already greater than or equal to n,
        # it is the smallest valid answer.
        if t >= n:
            return t

        i = n

        # Search for the first number whose digit product
        # is divisible by t.
        while prod(i) % t != 0:
            i += 1

        return i

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    // Returns the product of digits of a number.
    // If the number is a single digit, return it directly.
    int prod(int n) {
        if (n <= 9) {
            return n;
        }

        int prod = 1;
        while (n != 0) {
            prod *= (n % 10);
            n /= 10;
        }

        return prod;
    }
    int smallestNumber(int n, int t) {
        // If t is already greater than or equal to n,
        // it is the smallest possible answer.
        if (t >= n) return t;
        int i = n;
        // Keep checking consecutive numbers until
        // the product of its digits is divisible by t.
        while (prod(i) % t != 0) {
            i++;
        }
        return i;
    }
};